lib/Mojolicious/Plugin/AssetPack/Pipe/Fetch.pm

```diff
@@ -15,16 +33,20 @@ sub process {
   return $assets->each(
     sub {
       my ($asset, $index) = @_;
-      return unless $asset->format eq 'css';
+      return unless exists $formats{$asset->format} &&
+        exists $formats{$asset->format}->{regex} &&
+        exists $formats{$asset->format}->{subst}; 
       return unless $asset->url =~ /^https?:/;
 
       my $base    = Mojo::URL->new($asset->url);
       my $content = $asset->content;
+      my $regex = $formats{$asset->format}->{regex};
 
-      while ($content =~ /$URL_RE/g) {
+      while ($content =~ /$regex/g) {
         my ($pre, $url) = ($1, $2);
-        my $len   = length $url;
-        my $start = pos($content) - length($pre) - $len - 1;
+        my $len_pre   = length $pre;
+        my $len_url   = length $url;
+        my $start = pos($content) - $len_pre - $len_url - 1;
```

Comment:
Please tidy the code. There's a .perltidyrc file in the project directory you can use.